Gems/AtomTressFX/Code/Passes/HairPPLLResolvePass.cpp

```diff
@@ -0,0 +1,148 @@
+
+#include <Passes/HairPPLLResolvePass.h>
+#include <Rendering/HairFeatureProcessor.h>
+#include <Rendering/HairLightingModels.h>
+
+namespace AZ
+{
+    namespace Render
+    {
+        namespace Hair
+        {
+
+            HairPPLLResolvePass::HairPPLLResolvePass(const RPI::PassDescriptor& descriptor)
+                : RPI::FullscreenTrianglePass(descriptor)
+            {
+            }
+
+            void HairPPLLResolvePass::UpdateGlobalShaderOptions()
+            {
+                RPI::ShaderOptionGroup shaderOption = m_shader->CreateShaderOptionGroup();
+
+                m_featureProcessor->GetHairGlobalSettings(m_hairGlobalSettings);
+
+                shaderOption.SetValue(AZ::Name("o_enableShadows"), AZ::RPI::ShaderOptionValue{ m_hairGlobalSettings.m_enableShadows });
+                shaderOption.SetValue(AZ::Name("o_enableDirectionalLights"), AZ::RPI::ShaderOptionValue{ m_hairGlobalSettings.m_enableDirectionalLights });
+                shaderOption.SetValue(AZ::Name("o_enablePunctualLights"), AZ::RPI::ShaderOptionValue{ m_hairGlobalSettings.m_enablePunctualLights });
+                shaderOption.SetValue(AZ::Name("o_enableAreaLights"), AZ::RPI::ShaderOptionValue{ m_hairGlobalSettings.m_enableAreaLights });
+                shaderOption.SetValue(AZ::Name("o_enableIBL"), AZ::RPI::ShaderOptionValue{ m_hairGlobalSettings.m_enableIBL });
+                shaderOption.SetValue(AZ::Name("o_hairLightingModel"), AZ::Name{ "HairLightingModel::" + AZStd::string(HairLightingModelNamespace::ToString(m_hairGlobalSettings.m_hairLightingModel)) });
+                shaderOption.SetValue(AZ::Name("o_enableMarschner_R"), AZ::RPI::ShaderOptionValue{ m_hairGlobalSettings.m_enableMarschner_R });
+                shaderOption.SetValue(AZ::Name("o_enableMarschner_TRT"), AZ::RPI::ShaderOptionValue{ m_hairGlobalSettings.m_enableMarschner_TRT });
+                shaderOption.SetValue(AZ::Name("o_enableMarschner_TT"), AZ::RPI::ShaderOptionValue{ m_hairGlobalSettings.m_enableMarschner_TT });
+                shaderOption.SetValue(AZ::Name("o_enableDiffuseLobe"), AZ::RPI::ShaderOptionValue{ m_hairGlobalSettings.m_enableDiffuseLobe });
+                shaderOption.SetValue(AZ::Name("o_enableSpecularLobe"), AZ::RPI::ShaderOptionValue{ m_hairGlobalSettings.m_enableSpecularLobe });
+                shaderOption.SetValue(AZ::Name("o_enableTransmittanceLobe"), AZ::RPI::ShaderOptionValue{ m_hairGlobalSettings.m_enableTransmittanceLobe });
+
+                m_shaderOptions = shaderOption.GetShaderVariantKeyFallbackValue();
+            }
+
+            RPI::Ptr<HairPPLLResolvePass> HairPPLLResolvePass::Create(const RPI::PassDescriptor& descriptor)
+            {
+                RPI::Ptr<HairPPLLResolvePass> pass = aznew HairPPLLResolvePass(descriptor);
+
+                return AZStd::move(pass);
+            }
+
+            void HairPPLLResolvePass::InitializeInternal()
+            {
+                if (GetScene())
```

Comment:
As per my remark before, however, in this class' case the initialization might not depend on scene.  Since I'm not sure of that before running and testing properly under edge conditions (game launch, exiting, editor starts in parallel, etc..) I'm leaving it for now.
Let me know if you want to re-open it